### PR TITLE
Flow

### PR DESCRIPTION
**Pull request by nlepage**
### Description
First version of flow.

**Issue :** #133

### CODE REVIEW

**Comment by nlepage** [comment]
@frinyvonnick What do you think ? Not sure about the code arrangement.

**Comment by nlepage** [comment]
Hey @frinyvonnick @charlyx ! Like we talked about earlier today I managed to generate all of the flow exports of the curried functions.

**Comment by nlepage** [comment]
@frinyvonnick What do you think ? Should we merge as is and open a new issue to improve flow and avoid multiple reinstanciations ?

File: misc/generate-flow.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    const namespaces = _.keys(itemsByNamespace)
+    await Promise.all(namespaces.map(async namespace => {
+      const nsDir = path.resolve(flowDir, namespace)
+      await ensureDir(nsDir)
+
+      const nsItems = itemsByNamespace[namespace]
+
+      nsItems.forEach(async ({ name }) => await writeFile(
+        path.resolve(nsDir, `${name}.js`),
+        `import { ${name} } from '${namespace}/${name}'
```
Comment (frinyvonnick) [inline misc/generate-flow.js]: indentation is a bit weird here 😕 and in further generated code parts.

File: misc/generate-flow.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+const { curried } = ${name}
+
+export { curried as ${name} }
+`,
+      ))
+
+      await writeFile(
+        path.resolve(nsDir, 'index.js'),
+        `${nsItems.map(({ name }) => `import { ${name} } from './${name}'`).join('\n')}
```
Comment (frinyvonnick) [inline misc/generate-flow.js]: I think you can directly export all elements of the module here. Like :
```
 `export * from './${name}'`
```

File: misc/generate-flow.js (hunk 1 above)
Comment (nlepage) [reply misc/generate-flow.js]: I know but it allows to have a correct indentation in the generated files.
I haven't found a simple way of having a nicer indentation of the templates...
I think we can let it as it is and try to improve it later ?

**Pull request merged**